add PUT request for team/:id

elanals opened the pull request:
So that it's also possible to delete nested objects, as needed by the team settings UI

davidkokkelink reviewed (approved):
nice & simple

davidkokkelink commented:
before updating the UI to use this, we need to address the reason why this was made a `PATCH` (in cases where we set something for a customer that they cannot set themselves, so it doesn't get filtered out on a subsequent save)

elanals commented:
Changed this PR so that PUT requests preserve any existing protected settings, depends on changes to the FILTER function proposed in [this PR](https://github.com/datawrapper/plugin-subscriptions/pull/25).

Added a test case.

gka commented:
FYI: the original PATCH was implemented so that it replaced the entire settings. I changed that in a previous PR to be in line with other requests. Makes sense to add a PUT version.

davidkokkelink reviewed (approved):
nice tests
